micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -333,6 +337,33 @@ impl WifiCredentialStorage for NVSStorage {
     }
 }
 
+impl NetworkSettingStorage for NVSStorage {
+    type Error = NVSStorageError;
+    fn has_network_settings(&self) -> bool {
+        self.has_blob(NVS_NETWORK_SETTINGS_KEY).unwrap_or(false)
+    }
+
+    fn get_network_settings(&self) -> Result<Vec<NetworkSetting>, Self::Error> {
+        let blob: Vec<u8> = self.get_blob(NVS_NETWORK_SETTINGS_KEY)?;
+        let networks: Vec<NetworkSetting> = from_bytes(&blob).unwrap();
```

Comment:
TODO: map the err

Reply:
Is there a reason not to do this before opening the PR?

Reply:
I missed it when I opened the PR, and didn't want to push a change from under people if they were in the middle of reviewing cuz it can be annoying